src/core/occupation.jl

```diff
@@ -1,20 +1,79 @@
+import Roots
+using SpecialFunctions: erf, factorial
+
+
 """
-Compute the occupation at zero temperature and without smearing
+Compute the occupation without any smearing
 for `n_elec` electrons and the bands `Psi` with associated `energies`.
+Note: This function only provides physical occupations for insulators
+at zero temperature.
 """
-function occupation_zero_temperature(basis, energies, Psi, n_elec)
+function occupation_step(basis, energies, Psi, n_elec)
     n_bands = size(Psi[1], 2)
-    T = real(eltype(Psi[1]))
+    T = eltype(energies[1])
 
     @assert n_elec % 2 == 0 "Only even number of electrons implemented"
-    @assert n_bands ≥ n_elec / 2
+    n_occ = div(n_elec, 2)
+    @assert n_bands ≥ n_occ
 
     occupation = similar(basis.kpoints, Vector{T})
     for ik in 1:length(occupation)
         occupation[ik] = zeros(T, n_bands)
-        occupation[ik][1:Int(n_elec / 2)] .= 2
+        occupation[ik][1:n_occ] .= 2
     end
+
     occupation
 end
 
-# TODO Implement smearing
+# Ref for the equations:
+#    - M. Methfessel, A. T. Paxton 1989
+#       "High-precision sampling for Brillouin-zone integration in metals"
+#     - E. Cancès, V. Ehrlacher, D. Gontier, A. Levitt, D. Lombardi
+#       "Numerical quadrature in the brillouin zone for periodic schrodinger operators"
+# TODO: Marzari-Vanderbilt "cold smearing"
+smearing_fermi_dirac(x) = 1 / (1 + exp(x))
+smearing_gaussian(x) = (1 - erf(x)) / 2
+
+H1(x) = 2x
+H2(x) = 4x^2 - 2
+H3(x) = 8x^3 - 12x
+A_coeff(n) = (-1)^n / (factorial(n) * 4^n * sqrt(π))
+const A1 = A_coeff(1)
+const A2 = A_coeff(2);
+smearing_methfessel_paxton_1(x) = smearing_gaussian(x) + A1 * H1(x) * exp(-x^2)
+function smearing_methfessel_paxton_2(x)
+    smearing_gaussian(x) + A1 * H1(x) * exp(-x^2) + A2 * H3(x) * exp(-x^2)
+end
+
```

Comment:
I understand you don't like long lines, but in this case it kills off the symmetry with the previous line and makes it harder to see it's just one more term

Reply:
How about
```
smearing_methfessel_paxton_2(x) = smearing_methfessel_paxton_1(x) + A2 * H3(x) * exp(-x^2) 
```
That's shorter and makes it apparent, too?